app/assets/javascripts/darkswarm/controllers/credit_cards_controller.js.coffee

```diff
@@ -20,3 +20,7 @@ angular.module('Darkswarm').controller "CreditCardsCtrl", ($scope, $http, Credit
     ).finally ->
       window.location.reload()
 
+
+  $scope.hasOneDefaultSavedCards = () ->
+    $scope.savedCreditCards.length > 0 && $scope.savedCreditCards.some((card) -> card.is_default)
```

Comment:
Shouldn't it be disabled when there's no card?

From memory, that case is covered already but I don't know why this conditional is here.

Reply:
Not sure to understand actually.
 - If there's no card, then `hasOneDefaultSavedCards ` returns `false`. 
 - By the way, if there's no card, the "Allow charges" is hidden, via `hide: 'CreditCard.visible || savedCreditCards.length == 0'`

https://github.com/openfoodfoundation/openfoodnetwork/blob/825058472b5bd70a6f86ea872e3c5720f3d2489a/app/views/spree/users/_cards.html.haml#L21

Reply:
> If there's no card, then hasOneDefaultSavedCards returns false.

Yes, and would the result be the same if we did this?
```suggestion
    $scope.savedCreditCards.some((card) -> card.is_default)
```

Or even if the result was different, we wouldn't care. My point was that we don't need the length check, do we?

Reply:
I added this as commit and the specs are passing.